scripts/upload-source-maps.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+import { execSync } from 'child_process'
+
+const main = () => {
+	if (process.env.VERCEL_ENV === 'development') {
+		return
+	}
+
+	const LATEST_SHA = process.env.VERCEL_GIT_COMMIT_SHA
+	const PATH_PREFIX = 'https://developer.hashicorp.com/_next/static/'
+	const SERVICE = 'developer.hashicorp.com'
```

Comment:
I think we have two other possibilities when `process.env.VERCEL_ENV !== 'development'`:

- `preview`, for example for PR previews, branch deploys such as off our `staging` branch, etc
- `production`, the deployment that's served from `developer.hashicorp.com`

I noticed in DataDog we have two RUM applications set up, each with their own `SERVICE` identifier:

- `non-prod.developer.hashicorp.com`
- `developer.hashicorp.com`

Would it make sense to specifically target `developer.hashicorp.com` with `process.env.VERCEL_ENV === 'production'`, in order to reduce non-production noise in our production application? We could upload the other source maps to the `non-prod.developer.hashicorp.com` service.

Notably, I think we already have the necessary config to distinguish between the non-prod & prod services:

- [base.json config for non-prod](https://github.com/hashicorp/dev-portal/blob/70fb5226724a8cca166384645f88e26d950e3f26/config/base.json#L29)
- [production.json config](https://github.com/hashicorp/dev-portal/blob/70fb5226724a8cca166384645f88e26d950e3f26/config/production.json#L11)